Code/Common/include/sitkCompositeTransform.h

```diff
@@ -0,0 +1,204 @@
+ * SetNthTransformToOptimize* and SetAllTransformToOptimize* are used to
+ * set and clear flags arbitrarily. SetOnlyMostRecentTransformToOptimizeOn is
+ * a convenience method for setting only the most recently added transform
+ * for optimization, with the idea that this will be a common practice.
+ *
+ * Indexing:
+ * The index values used in GetNthTransform and
+ * SetNthTransformToOptimize* and SetAllTransformToOptimize* follow the
+ * order in which transforms were added. Thus, the first transform added is at
+ * index 0, the next at index 1, etc.
+ *
+ * Inverse:
+ * The inverse transform is created by retrieving the inverse from each
+ * sub transform and adding them to a composite transform in reverse order.
+ * The m_TransformsToOptimizeFlags is copied in reverse for the inverse.
+ *
+ * \ingroup ITKTransform
+ */
+class SITKCommon_EXPORT CompositeTransform
+    : public Transform {
+public:
+  using Self = CompositeTransform;
+  using Superclass = Transform;
+
+  /** \brief Construct an empty CompositeTransform
+   *
+   *  The created CompositeTransform is initialized with zero transforms.
+   *  Additional transforms of dimensions can be added.
+   */
+  explicit CompositeTransform(unsigned int dimensions);
+
+  /** \brief Create CompositeTransform converted or holding the transform argument.
+   *
+   * If the Transform is internally a CompositeTransform, a shallow copy to
+   * the internal transform will be made. Otherwise a new CompositeTransform is
+   * constructed which holds the transform argument.
+   */
+  CompositeTransform( const Transform &);
+
+  /** \brief A shallow copy constructor */
+  CompositeTransform( const CompositeTransform & );
+
+  ~CompositeTransform() override;
+
+  CompositeTransform &operator=( const CompositeTransform &);
+
+    /** Name of this class */
+  std::string GetName( ) const override  { return std::string("CompositeTransform");}
+
+
+  /** \brief Remove nested composite transforms
+   *
+   * If this transform is a composite which contains another nested
+   * composite transform, then the nested composite's transforms are
+   * placed into this transform. Nested composite transform may not be
+   * written to a file.
+   */
+  SITK_RETURN_SELF_TYPE_HEADER FlattenTransform();
+
+  /** \brief Add transform to the stack
+   *
+   * The transform is added to the back of the queue. The back of the queue is
+   * transforms points before the front.
+   */
+  SITK_RETURN_SELF_TYPE_HEADER AddTransform( Transform t );
+
+  /** \brief The number of transforms in the stack */
+  unsigned int GetNumberOfTransforms() const;
+
+  /** \brief Removed all transform from the stack. */
+  void ClearTransforms();
+
+  /** \brief Remove the last transform
+   *
+   * If the stack is empty and exception will be thrown.
+   */
+  void RemoveTransform();
+
+  /** \brief Get a shallow copy of the back transform
+   *
+   * If the stack is empty an exception will be thrown.
+   */
+  Transform BackTransform();
```

Comment:
I'd call this "TopTransform", using the stack terminology. I guess BackTransform is from the queue terminology (front and back of queue).

Reply:
Documentation has been updated.